### PR TITLE
{chem}[foss/2020b,fosscuda/2020b] NAMD v2.14

### PR DESCRIPTION
**Pull request by dithwick**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
@boegelbot: please test @ generoso

**Comment by SebastianAchilles** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 1 out of 2 (2 easyconfigs in total) cnx1 - Linux rocky linux 8.4, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz (haswell), Python 3.6.8 See https://gist.github.com/7851b440856e35d664e545d008b873a6 for a full test report.

`NAMD-2.14-fosscuda-2020b.eb` fails on generoso, because generoso has no GPUs.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3302.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), 4 x NVIDIA NVIDIA Tesla V100-SXM2-32GB, 465.19.01, Python 3.6.8
See https://gist.github.com/a2375482bf9e832399a0a931e9ce3d71 for a full test report.

File: easybuild/easyconfigs/n/NAMD/NAMD-2.14-foss-2020b-mpi.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    '34044d85d9b4ae61650ccdba5cda4794088c3a9075932392dd0752ef8c049235',  # NAMD_2.14_Source.tar.gz
+    'a838bd66a8f741247436687bd355c439f3d77542996d96e091b8b6bd0ccc34e2',  # NAMD-2.14-use_system_mremap_decl.patch
+]
+
+dependencies = [
+    ('Tcl', '8.6.10'),
+]
+
+# /bin/csh is required by 'config' script
+osdependencies = ['tcsh']
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/n/NAMD/NAMD-2.14-foss-2020b-mpi.eb:25]:
```suggestion
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]
I think it is better to use a `builddependencies` for `tcsh` instead of `osdependencies`.

File: easybuild/easyconfigs/n/NAMD/NAMD-2.14-fosscuda-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    'a838bd66a8f741247436687bd355c439f3d77542996d96e091b8b6bd0ccc34e2',  # NAMD-2.14-use_system_mremap_decl.patch
+]
+
+# support for GCC 8+ on POWER
+dependencies = [
+    ('Tcl', '8.6.10'),
+]
+
+# /bin/csh is required by 'config' script
+osdependencies = ['tcsh']
```
[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/n/NAMD/NAMD-2.14-fosscuda-2020b.eb:29]: @dithwick I've just updated the NAMD easyblock to make this work:
https://github.com/easybuilders/easybuild-easyblocks/pull/2654

[the remark this answers is not in the document]
Comment (dithwick) [reply easybuild/easyconfigs/n/NAMD/NAMD-2.14-fosscuda-2020b.eb:29]: Thanks, I'm just testing although I can't see why it shouldn't work.

**Comment by dithwick** [comment]
depends on easyblock PR:

https://github.com/easybuilders/easybuild-easyblocks/pull/2654

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/ed355b0f3c987463baf824e2853dc3d1 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
b-cn1126.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, 8 x NVIDIA Tesla K80, 470.129.06, Python 3.8.10
See https://gist.github.com/077a85b4ed29714fbcd0d17452958726 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
b-cn1126.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, 8 x NVIDIA Tesla K80, 470.129.06, Python 3.8.10
See https://gist.github.com/73444f236ef31815a0d06a377742175f for a full test report.

**Comment by akesandgren** [comment]
Unfortunately the changes in https://github.com/easybuilders/easybuild-easyblocks/pull/2654 is not enough.

Our current tcsh package does not have a csh cmd, only a tcsh. And the builddep on tcsh is not enough at the moment and needs to be a runtime dep since there a re plugin scripts in NAMD that uses (t)csh

**Review by akesandgren — CHANGES_REQUESTED** [review]

**Comment by akesandgren** [comment]
Updated tcsh easyconfigs in #15571

File: easybuild/easyconfigs/n/NAMD/NAMD-2.14-foss-2020b-mpi.eb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+source_urls = ['https://www.ks.uiuc.edu/Research/namd/%(version)s/download/946183/']
+sources = ['%(name)s_%(version)s_Source.tar.gz']
+patches = ['%(name)s-%(version)s-use_system_mremap_decl.patch']
+checksums = [
+    '34044d85d9b4ae61650ccdba5cda4794088c3a9075932392dd0752ef8c049235',  # NAMD_2.14_Source.tar.gz
+    'a838bd66a8f741247436687bd355c439f3d77542996d96e091b8b6bd0ccc34e2',  # NAMD-2.14-use_system_mremap_decl.patch
+]
+
+builddependencies = [
```
[the remark this answers is not in the document]
Comment (dithwick) [reply easybuild/easyconfigs/n/NAMD/NAMD-2.14-foss-2020b-mpi.eb]: I've made the changes to the easyconfigs

**Comment by dithwick** [comment]
Apologies, completely forgot about submitting this. I've made the requested changes.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
b-cn1501.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, 2 x NVIDIA Tesla V100-PCIE-16GB, 470.161.03, Python 3.8.10
See https://gist.github.com/e40fb296b5ad7bdcb356e861efe5934f for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
b-cn1501.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, 2 x NVIDIA Tesla V100-PCIE-16GB, 470.161.03, Python 3.8.10
See https://gist.github.com/b3926c6bf6c077873f8336c6e28fd0ca for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @dithwick!

**Pull request merged**